python/requirements_compiled.txt

```diff
@@ -1519,7 +1519,7 @@ py-cpuinfo==9.0.0
     # via deepspeed
 py-partiql-parser==0.5.0
     # via moto
-py-spy==0.3.14
+py-spy==0.4.0 ; python_version < "3.12"
```

Comment:
so we just upgrade py-spy to 0.4.0, like for images of all python versions now?

Reply:
nice!

wonder why we did not upgrade `py-spy` earlier.. like was there something blocking us to do so? seems like a simple change based on this PR..

Reply:
Just wasn't aware of 0.4.0's release. Which was released this november. I just saw someone comment that 0.4.0 had python 3.12 support so I updated after seeing that.